firedrake/slate/slac/tsfc_driver.py

```diff
@@ -47,10 +60,12 @@ def compile_terminal_form(tensor, prefix=None, tsfc_parameters=None):
     cxt_kernels = []
     for orig_it_type, integrals in transformed_integrals.items():
         subkernel_prefix = prefix + "%s_to_" % orig_it_type
-        kernels = tsfc_compile(Form(integrals),
+        form = Form(integrals)
+        kernels = tsfc_compile(form,
                                subkernel_prefix,
                                parameters=tsfc_parameters)
         cxt_k = ContextKernel(tensor=tensor,
```

Comment:
Yeah. When I originally introduced the `ContextKernel`, it was mainly meant to help organize the incoming TSFC kernels corresponding to a particular form. And it happens to be the case that a `Tensor` can have many integrals contained in `Tensor.form` (true also if the form is mixed), hence why multiple context kernels share the same Slate tensor. I'm debating whether or not to change it in this PR. I think it could work just fine with one `ContextKernel` per `Tensor`, but I need to think about how to organize the various integral types. Thoughts?